soccer/gameplay/skills/capture.py

```diff
@@ -49,9 +54,23 @@ def __init__(self, faceBall=True):
 
         self.add_transition(
             Capture.State.fine_approach,
-            behavior.Behavior.State.completed, lambda: self.bot_near_ball(
+            Capture.State.delay, lambda: self.bot_near_ball(
                 constants.Robot.Radius + constants.Ball.Radius), 'has ball')
 
+        self.add_transition(
+            Capture.State.delay, 
+            behavior.Behavior.State.completed, 
+            lambda: time.time() - self.start_time > Capture.DelayTime and self.bot_near_ball(
+                constants.Robot.Radius + constants.Ball.Radius),
```

Comment:
Would using has_ball() make more sense here than a proximity check? Since has_ball() checks if the ball is actually detected by the mouth of the robot.

Reply:
the ball sensor is actually quite unreliable on the robots currently (the testing bot we have has it always on). We really need to build a more generic function which uses ball sense only if it's available, and disables using it if we think it's broken.

Reply:
At the very least, I think we should be checking if the robot is facing the ball here, not just if the ball is touching the robot